src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackDecoder.cs

```diff
@@ -249,6 +249,11 @@ public void Decode(ReadOnlySpan<byte> data, bool endHeaders, HeaderCallback onHe
 
                         if (_integerDecoder.StartDecode((byte)(b & ~HuffmanMask), StringLengthPrefix))
                         {
+                            if (_integerDecoder.Value == 0)
+                            {
+                                throw new HPackDecodingException(SR.Format(SR.net_http_invalid_response_header_name, ""));
+                            }
+
```

Comment:
Is there a test to go along with this?

Reply:
Seems like it was in master: https://github.com/dotnet/corefx/pull/39907/files#diff-65fa6f9cff4a3dd065a4b439b2c13cb7R270